### PR TITLE
Raul pedro redesign reports page

### PR DESCRIPTION
**Pull request by rauleffting**
# Description
(REPORTS COMPONENT - PRIORITIES HIGH) - **Continuation of [PR 641](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/641)**

image

## Mainly changes explained:
The reports page was redesigned. Now, it's possible to get data from teams and shows it with charts. It's possible to search teams by name etc.

## How to test:
1. git checkout raul_pedro_redesign_reports_page
2. npm install
3. npm run start:local
4. log in as admin.
5. Reports -> Teams -> Choose a team (the one with my name has data, so it's better for testing).
6. Test all features on the reports page (searching, selecting, etc.) and ensure it all works.

### Here is a video to help you:

https://user-images.githubusercontent.com/29555732/229852623-00fab088-047c-43cf-9947-cc890a49eee5.mp4

### CODE REVIEW

**Comment by AnishPandita** [comment]
Hi @rauleffting I thoroughly tested your Pr and it seems to be working great. Looks visually appealing too.

https://user-images.githubusercontent.com/111321945/230006235-7412207b-fe3d-4d81-a4ac-a4ced1e50663.mov

 I noticed in this particular case, when i scrolled down completely and opened the see dropdown the menu opened towards bottom and was cut but when i scrolled up it was opening properly on upper side.

https://user-images.githubusercontent.com/111321945/230007066-cb2ee43b-86c0-447b-a634-d6b1b98d0348.mov

**Review by ToobaJamal — DISMISSED** [review]
Hi Raul, I tested the changes and it looks good on my end. \

\

\

\

**Comment by Elisaudo** [comment]
Hi Raul, I tested the changes and the new design is great. Like shown in the video, I detected 3 issues in the test.
1)	In team report, the report works with the Options “Manager and Admin”. With the option “Owner” the screen stays white and page crashes.
2)	In team report, When scrolling down, the calendar disappears in part.
3)	In Project report the page crashes when click on the previous arrow on the browser, after Search and opens a Project page.

https://watch.screencastify.com/v/Lnuo7Flu4Ffhjdu4fynh

I suggest that the HGN color palette be revised or updated so that all pages are within the same pattern, to be followed by the developers.

**Review by ypan710 — COMMENTED** [review]
@rauleffting Hi Raul! I've tested your PR and most of the features worked as expected, but I'm not sure if clicking on the "See" button under the "Team Members" column to access the different team member's profile is part of this PR's expected functionalities.

Screen Shot 2023-04-05 at 11 04 07 PM

**Comment by one-community** [comment]
Chiming in here that we want to keep this color pattern. It's different than the dashboard but the same as what we'll be using for all of phase 2 and future phases. This color palette will differentiate deeper levels of access and interaction within the app.

**Review by vitorsaa2k — COMMENTED** [review]
When selecting a team that have no members, it says that the team have 1 member, as showed on the image below.

image

despite that everything works fine.

**Comment by rauleffting** [comment]
> Hi Raul, I tested the changes and the new design is great. Like shown in the video, I detected 3 issues in the test.
> 
> 1. In team report, the report works with the Options “Manager and Admin”. With the option “Owner” the screen stays white and page crashes.
> 2. In team report, When scrolling down, the calendar disappears in part.
> 3. In Project report the page crashes when click on the previous arrow on the browser, after Search and opens a Project page.
> 
> I suggest that the HGN color palette be revised or updated so that all pages are within the same pattern, to be followed by the developers.

Hi, @Elisaudo. You are viewing an outdated version of the PR. Please try to do 'git pull'. Thank you very much!

**Comment by rauleffting** [comment]
Hi @vitorsaa2k. Thanks for your review. The chart isn't being displayed because the team member doesn't have hours to be displayed.

**Comment by rauleffting** [comment]
Hi @nicollext, this is because it hasn't any inactive teams to be displayed. Thank you very much!

File: src/components/Reports/TeamReport/components/UserLoginPrivileges.jsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    });
+  }, [selectedTeamsData]);
+
+  useEffect(() => {
+    const selectedTeamsMembersArray = selectedTeams.map((team) => allTeamsMembers[team.index]);
+    setSelectedTeamsMembers(selectedTeamsMembersArray);
+  }, [selectedTeams, allTeamsMembers]);
+  
+  // Check if the user has admin privileges
+  if (role == ('Administrator' || 'Owner')) {
```
Comment (EvianTan) [inline src/components/Reports/TeamReport/components/UserLoginPrivileges.jsx:79]: Please change the above check to` if (role === `Administrator` || role === `Owner` )`
Your current check only passes when the role is admin.... I used my owner account, got totally blank page there

**Review by EvianTan — APPROVED** [review]
Fantastic work!

**Pull request merged**